### ISSUE TITLE
[0035] DXIL Validator should catch undef Matrix uses

### ISSUE THREAD

**Issue by anupamachandra** [microsoft/DirectXShaderCompiler#8464]
The LinAlg [spec](https://github.com/microsoft/hlsl-specs/blob/main/proposals/0035-linalg-matrix.md) doesn't disallow unitialized LinAlg Matrices to be used in Matrix operations. So the code snippet below:
```cpp
MatrixTy mat;
Buffer.Store(0, mat.Get(0));
```

produces:
```
 %3 = call half @dx.op.linAlgMatrixGetElement.f16.mC8M16N16U0S2(i32 -2147483630, ** %dx.types.LinAlgMatrixC8M16N16U0S2 undef**, i32 0)  ; LinAlgMatrixGetElement(matrix,threadLocalIndex)
```
Proposed Solution:
Improve user experience by haivng DXIL Validator catch this and emit errors.

**Comment by V-FEXrt** [comment]
This is done now! Was part of https://github.com/microsoft/DirectXShaderCompiler/pull/8588
